runtime/bindings/python/hal.cc

```diff
@@ -1283,11 +1287,11 @@ void SetupHalBindings(nanobind::module_ m) {
 
   m.def(
       "get_cached_hal_driver",
-      [driver_cache](std::string device_uri) {
+      [driver_cache](std::string device_uri, std::optional<bool> clean) {
```

Comment:
You don't need to wrap a bool in an optional. Can just default it to false instead of py::none().

Reply:
Ah.  I was looking for another example of an optional bool in the API and didn't find one.  I'm a nanobind n00b, so I'll have to figure this one out.